dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerAPIImpl.java

```diff
@@ -811,6 +812,140 @@ public Container save(Container container, List<ContainerStructure> containerStr
 		return container;
 	}
 
+	@WrapInTransaction
+	@Override
+	@SuppressWarnings("unchecked")
+	public Container save(Container container, List<ContainerStructure> containerStructureList,

```

Comment:
all params should be final to avoid accidental modifications

Reply:
The ContainerForm is a view object we do not used to pass it to the API layer, since may be context were it not exists.
That said, you should pass a Container filled with the ContainerForm in advance

Reply:
why do we need a new save method? can we just use this one

https://github.com/dotCMS/core/blob/90f745e1d2fc266b1e35e36168a3c9f46251a603/dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerAPIImpl.java#L703

... or overwrite it

Reply:
Really appreciate your feedback - code adjustments are done and new method **save** has been removed and also I have marked all the parameters as final or the existing method **save** 

@jdotcms @freddyucv

Reply:
Please have a look at the recent commit - thanks